### ISSUE TITLE
Implement UI for new AdSense setup account components

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#4763]
This issue is for actually implementing the UI (and a bit off extra logic) for the enhanced AdSense setup flow main components that were introduced in #4759.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `SetupAccountNoClient` component from #4759 should have its UI implemented as follows:
    * It should be a modified version of the existing (V1) `SetupAccountNoClient` component (**not reusing the existing component**), with the following tweaks:
        * The heading should be replaced with: _Looks like you need to upgrade your AdSense account_
        * The main body text should be replaced with: _To start using AdSense on your website, you need to upgrade your account to add “AdSense for content”._, followed by a _Learn more_ link that points to the same support URL currently used.
        * The current link at the bottom of the page (which is now inline right after the body text, see above) should be replaced with a CTA button:
            * Button text: _Apply now_
            * Button URL: _https://www.google.com/adsense_
            * Clicking the button should fire a GA tracking event (category: `${ viewContext }_adsense`, name: `apply_afc`), and the link URL should open in a new window.
* The `SetupAccountCreateSite` component from #4759 should have its UI implemented as follows:
    * It should be a modified version of the existing (V1) `SetupSiteAdd` component (**not reusing the existing component**), with the following tweaks:
        * The heading should be replaced with: _Add this site to your AdSense account_
        * Clicking the CTA button (which should otherwise not be modified) should fire a GA tracking event (category: `${ viewContext }_adsense`, name: `create_site`) before opening the URL.
* The `SetupAccountPendingTasks` component from #4759 should have its UI implemented as follows:
    * It should be a new component, but it's a good idea to use a copy of the existing (V1) `SetupSiteAdd` component as a starting point (e.g. for the markup, which can be reused). The contents should be entirely changed, based on the following:
        * The heading should be: _Your account isn’t ready to show ads yet_
        * The main body text should be: _You need to fix some things before we can connect Site Kit to your AdSense account._
        * Right below that paragraph, the component should include the `SetupUseSnippetSwitch` from #4761.
        * The CTA button should be as follows:
            * Button text: _Review AdSense account_
            * Button URL: the result of `getServiceAccountURL()` selector
            * Clicking the button should fire a GA tracking event (category: `${ viewContext }_adsense`, name: `review_tasks`), and the link URL should open in a new window.
* Small aside: The `getServiceAccountURL` selector in the `modules/adsense` store should be modified to no include the `/home` suffix in the URL _if_ the `adsenseSetupV2` feature flag is enabled.
* The UI implemented here can be tested via the Storybook stories introduced in #4759, and it should be ensured it looks correct there (in case e.g. any additional data needs to be bootstrapped in the Storybook configuration).

## Implementation Brief
* Using `assets/js/modules/adsense/components/setup/v2/SetupAccountNoClient.js`,
	* Copy over the code from `assets/js/modules/adsense/components/setup/SetupAccountNoClient.js` with the text changes as per the AC and the following changes:
		* Replace the `Link` component by `Button` with the details from the AC.
		* Create a new function `onButtonClick` which ese `trackEvent` for tracking GA events as per the AC. The `onButtonClick` function should be the value for the `onClick` prop of `Button`.
* Using `assets/js/modules/adsense/components/setup/v2/SetupAccountCreateSite.js`,
	* Copy over the code from `assets/js/modules/adsense/components/setup/SetupSiteAdd.js` with the text changes as per the AC and the following changes:
		* Update `addSiteHandler` to call `trackEvent` with parameters as per the AC before calling `global.open`.
* Using `assets/js/modules/adsense/components/setup/v2/SetupAccountPendingTasks.js`,
	* Copy over the code from `assets/js/modules/adsense/components/setup/SetupSiteAdd.js` focusing mainly on keeping the markup with the text changes as per the AC and the following changes:
		* Discard existing selector calls and `addSiteHandler` function.
		* Query the `modules/adsense` data store via the `getServiceAccountURL` to get the service account URL which will be used for the CTA at the bottom.
		* Define a new `onButtonClick` function which should call `trackEvent` with parameters as per the AC and will be the value for the `onClick` prop of `Button`.
		* Include `assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js` below the `p` tag.
* Using `assets/js/modules/adsense/datastore/service.js`,
	* Update `getServiceAccountURL` to include the `/home` suffix in the URL only if the `adsenseSetupV2` is not enabled (via the `isFeatureEnabled` function).
* Ensure the stories in `assets/js/modules/adsense/components/setup/v2/SetupMain.stories.js` are rendered correctly for the above variants.

### Test Coverage
* No new tests to be added.

## QA Brief

- Ensure you are using the latest Tester plugin (v1.8.1 at the time of writing).
- Enable the `adsenseSetupV2` feature flag.

### No Client

- Create a new site (or use an existing one which _has not_ previously had the AdSense module set up).
- Using the Tester plugin, force AdSense **account** status to `no-client`.
- Start the AdSense setup process, via **Settings** > **Connect more services** > **Set up AdSense**.
- Verify the "No Client" flow is displayed, i.e. the `SetupAccountNoClient` component as seen in the **Account: No Client** story (see below).

### Create Site

- Using the Tester plugin, force AdSense **site** status to `none`.
- Start the AdSense setup process, via **Settings** > **Connect more services** > **Set up AdSense**.
- Verify the "Create Site" flow is displayed, i.e. the `SetupAccountCreateSite` component as seen in the **Account: Create Site** story (see below).

### Pending Tasks

- Using the Tester plugin, force AdSense **account** status to `needs-attention`.
- Start the AdSense setup process, via **Settings** > **Connect more services** > **Set up AdSense**.
- Verify the "Pending Tasks" flow is displayed, i.e. the `SetupAccountPendingTasks` component as seen in the **Account: Pending Tasks** story (see below).
  - Note, though, that the use snippet toggle won't be present, this needs an existing AdSense tag to be present in order to display.
- Repeat the above with the forced AdSense account statuses `client-requires-review` and `client-getting-ready`.`
- Using the Tester plugin's **Place AdSense existing tag** feature, mock an existing tag, and repeat the above, verifying the use snippet toggle does appear.

### Storybook

- Check the relevant **AdSense Setup V2** stories in Storybook to ensure the components render as expected with content as described in the AC.
  - **Account: No Client** for the `SetupAccountNoClient` component
  - **Account: Create Site** for `SetupAccountCreateSite`
  - **Account: Pending Tasks** for `SetupAccountPendingTasks`
![image](https://user-images.githubusercontent.com/18395600/165545208-facc76f7-8168-425a-b818-bd4c2cda8a0a.png)

<details>
<summary>
As there is no design to refer to, here are some screenshots for reference:
</summary>

**Account: No Client**
![image](https://user-images.githubusercontent.com/18395600/165546423-c2c9810b-7a98-424d-bff5-b07ee2c3fb73.png)

**Account: Create Site**
![image](https://user-images.githubusercontent.com/18395600/165546521-0c90a9b4-1cd6-4ab1-bb6c-976d54ead3d3.png)

**Account: Pending Tasks**
![image](https://user-images.githubusercontent.com/18395600/167145812-9f674ead-e1a1-44db-9836-1acd1b21cb39.png)

</details>

## Changelog entry

* Add UI for new AdSense setup flow.

**Comment by tofumatt** [comment]
IB ✅

**Comment by techanvil** [comment]
Thanks for spotting that @mohitwp! I'm creating a followup PR to fix the button text.

> Question : Currently I'm not able to test flow for 'SetUpAccountNoClient' through tester plugin. Is there any other way by which I can test this condition ?

I've looked into this and discovered that it _is_ possible to test via the Tester plugin. I've amended the instructions in the QAB under **No Client**. Essentially you need to make sure to force account status `no-client` in the Tester plugin _before_ you start the AdSense setup process for the first time on a site, otherwise an AdSense `clientID` can become set and cached in the AdSense settings.
